libbeat/publisher/pipeline/retry.go

```diff
@@ -126,10 +132,12 @@ func (r *retryer) loop() {
 
 		log = r.logger
 	)
+	r.doneWaiter.Add(1)
```

Comment:
move the add before the go-routine is started, not within the go-routine. It is the constructor that starts the loop.